probe-rs/src/core/memory_mapped_registers.rs

```diff
@@ -0,0 +1,99 @@
+/// A memory mapped register, for instance ARM debug registers (DHCSR, etc).
+pub trait MemoryMappedRegister<T>: Clone + From<T> + Into<T> + Sized + std::fmt::Debug {
+    /// The register's address in the target memory.
+    /// For some architectures (e.g. ARM Cortex-A) this address is offset from a base address.
+    /// For others (e.g. ARM Cortex-M, RISC-V) this address is absolute.
+    const ADDRESS_OFFSET: u64;
+    /// The register's name.
+    const NAME: &'static str;
+    /// Get the register's address in the memory map.
+    /// For some architectures (e.g. ARM Cortex-A) this address is offset from a base address.
+    /// For others (e.g. ARM Cortex-M, RISC-V) this address is absolute.
+    fn get_mmio_address(base_address: Option<u64>) -> u64 {
```

Comment:
Could this be solved with two separate methods? 

This is only relevant for Cortex-A as far as I know, so you end up with loads of `get_mmio_address(None)`, and I find this pretty unreadable. It's not clear what `None` means in this case.

Reply:
Fair comment. I will change it.